### ISSUE TITLE
Volto 16/17 cannot connect to backend with Node 18, but works fine with Node 16

### ISSUE THREAD

**Issue by fredvd** [plone/volto#5168]
This is a a strange bug, only me and @ericof have spotted in the wild so far.  

I have a Volto project with Volto 17 alpha set up with Node 16 that works fine, but as soon as I switch the project to Node 18, the frontend can no longer connect to the backend. This is strange, as the backend is available on http://localhost:8080/Plone and even http://localhost:3000/++api++ shows the correct responses/json object from the backend !!

@davisagli remarked early on this could have something to do with the ipv6 support in Node 18 where dns resolution has been changed.  But using 127.0.0.1 instead of localhost in combination with RAZZLE_DEV_PROXY_API_PATH doesn't solve the issue. 

Even serving the backend from port 9000, the frontend from port 4000 and the api path change above, gives the same result. 

To debug this further I digged into the ipv6/ipv4 changes in node 17/18, one of the changes is also dns resolution order that now prefers ipv6. Using 127.0.0.1 instead of 'localhost' side tracks this change.  But you can change the default back by calling 
[dns.setDefaultResultOrder](https://nodejs.org/dist/latest-v17.x/docs/api/dns.html#dnssetdefaultresultorderorder)

So that's what I did, I added
```
import dns from 'dns';
dns.setDefaultResultOrder('ipv4first');
```

to the top of https://github.com/plone/volto/blob/f37e8c12f5e2cc930344cdcb3e85e71710b4c6db/src/start-server.js#L1-L7

And miraculously my backend is found also under Node 18 by the frontend.  !!!

The ConnectionRefused error view is shown based on redux connect state.apierror in App.jsx here: 

https://github.com/plone/volto/blob/f37e8c12f5e2cc930344cdcb3e85e71710b4c6db/src/components/theme/App/App.jsx#L327

so I guess something in redux is stlil hardcoded to check localhost and returns the ECONNREFUSED in the state, even if we configure the api path to be 127.0.0.1 where ipv4 should be used and the dns resolution order should be irrelevant.

**Comment by davisagli** [comment]
@fredvd What command are you using to start volto? And, can I look at the repository for your project?

I'm thinking, though, that we should probably add `dns.setDefaultResultOrder('ipv4first');` to volto core.

**Comment by sneridagh** [comment]
@fredvd good catch! +1 to add the line to choose the priority!

**Comment by sneridagh** [comment]
The API layer only sets the state of `connectionRefused` to true in here:

https://github.com/plone/volto/blob/a0a89be0deb467c320bc0b2be69af4caece69962/src/middleware/api.js#L281-L291

and only if "ECONNREFUSED" comes from the network layer in the server (can't come from the browser). So it happens in how Node handles the network connection and issues an error. Definitely, they changed something from 18.

**Comment by fredvd** [comment]
@davisagli  The project is generated from the plone 6 cookiecutter starter project from the offical docs (https://github.com/collective/cookiecutter-plone-starter), Added volto-light-theme and project specific frontend add'on as with some other kitconcept projects.

I start the frontend with yarn start,  which is "razzle start" in the package.json To not have to change start-server.js in the volto code which is in this project in node_modules, I changed [frontend-dir]/src/index.js to include the 2 lines to set the resolution to ipv4 first. 

---

To make sure that it isn't something custom in the project I just set up a clean project locally with https://github.com/collective/cookiecutter-plone-starter and followed the README, with USE_VOLTO_ALPHA to use Volto 17.0.0alpha25  .  To use the correct Python and node versions, I use volta.sh to pin the frontend directory to node 18.17.1 and use pyenv  to have Python 3.11.5 used in the backend directory. Both come from homebrew, M1 mac, OS X Monterey.

But result is the same, after starting both frontend and backend, creating a default Plone site in the backend, the frontend cannot connect to the backend. I add the 2 lines to src/index.js , restart the frontend and it can connect. It is unrelated to VLT or any other customisations.

**Comment by pnicolli** [comment]
>```javascript
>import dns from 'dns';
>dns.setDefaultResultOrder('ipv4first');
>```

If I understand correctly, adding these lines would just change the order that node follows to resolve names to be ipv4 first and ipv6 second. I am ok to add this if it fixes a problem, but I am not a huge fan of giving priority to an "old" standard in production servers, even if ipv6 will eventually be used if needed.
The reason why it scares me a little is because we can easily forget it's there in a few years if ipv4 starts dying (and I'm not sure it actually ever will).
Should we maybe start by adding this only when in development mode?

**Comment by sneridagh** [comment]
@pnicolli I would only add it in dev mode (and in the proxy middleware?)

**Comment by fredvd** [comment]
If you experience this issue and/or was referred to it, an easier place to add the two lines is in a scaffolded frontend' project directory its src/index.js 👍 

```
/**
 * Replace with custom runner when needed.
 * @module index
 */

import dns from 'dns';
import start from '@plone/volto/start-server';

dns.setDefaultResultOrder('ipv4first');

const reloadServer = start();

if (module.hot) {
  reloadServer();
}
```

**Comment by fredvd** [comment]
PR happening here: https://github.com/plone/volto/pull/5261

**Comment by mauritsvanrees** [comment]
I confirm the fix helps for me.

**Comment by davisagli** [comment]
Fix released in volto 17.0.1.
